README.md

```diff
@@ -58,42 +95,235 @@ import ReactDOM from 'react-dom';
+  - Type: {boolean}
+  - Default: false
+* `store.disableLoading` - flag of whether disable loading effect globally. Note, action's `disableLoading` flag will always takes priority when both are set
+  - Type: {boolean}
+  - Default: false
+
+#### Example
+
+```javascript
+const todos = store.useStore('todos');
+
+useEffect(() => {
+  todos.refresh();
+}, []);
+
+return (
+  <div>
+    {todos.refresh.error ? 
+      todos.refresh.loading ? (
+	    <div>
+	      loading.......
+	    </div>
+      ) : (
+        <div>
+	      error.......
+	     </div>
+      ) 
+    : (
+       <ul>
+         {dataSource.map(({ name }) => (
+           <li>{name}</li>
+         ))}
+       </ul>
+    )}
+  <Loading />
 );
 ```
+## Testing
 
-## 示例
+Because all the state and action are cleanly contained in a plain object, it's easy to write test without mocking.
 
-- Todos：[Sandbox](https://codesandbox.io/s/2017600okp)
+Example:
 
-## 反馈
+```javascript
+describe("todos", () => {
+  test("refresh data success", async () => {
+    await todos.refresh();
+    expect(todos.dataSource).toEqual([
+      {
+        name: "react"
+      },
+      {
+        name: "vue",
+        done: true
+      },
+      {
+        name: "angular"
+      }
+    ]);
+  });
+});
+```
 
-- [issues](https://github.com/alibaba/ice/issues)
+Please refer to the `todos.spec.js` file in the above sandbox for complete reference.
 
-## 参考
+## Best Practice
+
+### Never mutate state outside actions
+
+icestore enforce all the mutations to state can only ouccr in action methods. It will throw error when mutate state directly outside actions(eg. in view component). It will cause your mutation logic hard to trace and imposible to test if let mutation scatterd around view component other than store.
+
+```javascript
+  // store.js
+  export default {
+    inited: false,
+    setInited() {
+      this.inited = true;
+    }
+  }
+  
+  // view.js
+  const todos = useStore('todos');
+  
+  useEffect(() => {
+    // bad
+    todos.inited = true;
+    
+    // good
+    todos.setInited();
+  });
+```
+
+### Divide store as small as possible
+
+By design, `icestore` will trigger the rerender of all the view component subscribed to the store by `useStore` once the state of the store has changed. Which means puting more state in one store may cause more view compoenent to rerender which will affect the performance of app. So from performance aspect, it's advised to categorize your state and put them in indivisual stores. 
+
+### Don't overuse icestore
+
+From project management aspect, global store should only be used to store the state sharing ac ross multiple pages or state. Put local state in store will break component's self encapsulation which will affect its reusability. Take above `todo` app for example, if it exists as one page or component in the project, the state of `todo` app is prefered to store as local state in view component other than in global store.
```

Comment:
From the engineering perspective, the global store should only be used to store states that are shared across multiple pages. 

Putting local state in the global store will break the component's encapsulation, affecting its reusability. Using the `todo` app as an example, since the app only has one page, the state of the `todo` app is preferred to be stored as a local state in the view component rather than in the global store.